app/grandchallenge/components/forms.py

```diff
@@ -151,8 +165,60 @@ def _get_image_field(self, *, interface, current_value):
         )
 
     def _get_file_field(self, *, interface, values, current_value):
-        return self._get_default_field(
-            interface=interface, current_value=current_value
+        if current_value:
+            if isinstance(current_value, ComponentInterfaceValue):
+                # this happens on initial form load when there already is a CIV for
+                # the interface
+                return self._get_select_upload_widget_field(
+                    interface=interface,
+                    values=values,
+                    current_value=current_value,
+                )
+            else:
+                # on form submit, current_value either is the pk of an existing CIV
+                # or the UUID of a new UserUpload object
+                try:
+                    uuid.UUID(current_value)
+                    return self._get_default_field(
+                        interface=interface, current_value=current_value
+                    )
+                except ValueError as e:
+                    if ComponentInterfaceValue.objects.filter(
+                        pk=current_value
+                    ).exists():
```

Comment:
You're looking at **any** CIV here? Thinking data security, should it be filtered by permission or by membership of the object they definitely have permission to view/change?

Reply:
Hm, good point. That would be quite an expensive query though, right? I think it's not strictly necessary because we check that the value is part of the queryset (the `values` parameter that we pass to the widget) and we retrieve that from the DB in `base_obj.values_for_interfaces`.

Reply:
ah yes of course, don't know why I didn't see that earlier!